prax.c

```diff
@@ -201,29 +201,41 @@ static void *make_nl_req(int req, profile_t *process)
     struct sockaddr_nl addr;
     memset(&addr, 0, sizeof addr);
     addr.nl_family = AF_NETLINK;
-
-    int msglength = msg.nl.nlmsg_len;
-    char *msg_buffer = (char *) &msg;
+    
+    int msglength = msg->nl.nlmsg_len;
+    char *msg_buffer = (char *) msg;
 
     while (msglength > 0) {
         int bytes_sent = sendto(process->nl_conn, msg_buffer, msglength, 0,
                             (struct sockaddr *) &addr, sizeof addr);
         if (bytes_sent < 0)
-            return NULL;
+            goto release;
         msglength -= bytes_sent;
         msg_buffer += bytes_sent;
     }
+    
+    memset(msg, 0, sizeof *msg);
+    
+    if (recv_nl_req(process->nl_conn, msg) < 0)
+        goto release;
 
-    memset(&msg, 0, sizeof msg);
-    if (recv_nl_req(process->nl_conn, &msg) < 0)
-        return NULL;
 
-    void *task_ret = NULL;
+    if (req == TASKSTATS_CMD_GET) {
+        void *parse_results = parse_taskmsg(TASKSTATS_TYPE_STATS, msg);
+        if (parse_results) {
```

Comment:
Check for `NULL` and depending on the request allocate a block to cast and return